src/Compilers/Core/Portable/InternalUtilities/SemaphoreSlimExtensions.cs

```diff
@@ -16,8 +16,13 @@ public static SemaphoreDisposer DisposableWait(this SemaphoreSlim semaphore, Can
 
         public async static Task<SemaphoreDisposer> DisposableWaitAsync(this SemaphoreSlim semaphore, CancellationToken cancellationToken = default)
         {
-            await semaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
-            return new SemaphoreDisposer(semaphore);
+            return await semaphore.WaitAsync(cancellationToken)
+                .ContinueWith(
+                    (t, state) => { return new SemaphoreDisposer((SemaphoreSlim)state) },
```

Comment:
nit: `=> new SemaphoreDisposer...`

Reply:
nit: `_` for `t`.

Reply:
I tried, but that breaks type inference between `Action<T>` and `Func<T, TResult>`